mbed/mbed.py

```diff
@@ -1447,6 +1440,22 @@ def ignore_build_dir(self):
             except IOError:
                 error("Unable to write build ignore file in \"%s\"" % os.path.join(build_path, '.mbedignore'), 1)
 
+    def detect_target(self, info=None):
```

Comment:
There is a valid case where you have multiple platforms plugged in, but all of them are different. This could be supported by supplying an optional target name to this function. Shouldn't be too hard to implement:

```python
def detect_target(self, target_name=None, info=None):
    targets = self.get_detected_targets()

    if target_name:
        targets = [t for t in targets if t['name'].upper() == target_name.upper()]

    ...
```

(^Note that actual code is totally untested, but the idea should be sound)

But this can always be added in another PR. @screamerbg totally up to you on the priority of this.